src/definitions.jl

```diff
@@ -561,3 +576,58 @@ Pre-plan an optimized real-input unnormalized transform, similar to
 the same as for [`brfft`](@ref).
 """
 plan_brfft
+
+##############################################################################
+
+region(p::Plan) = p.region
+region(p::ScaledPlan) = region(p.p)
+
+# Projection style (:none, :real, or :real_inv) to handle real FFTs
+function projection_style end
+# Length of halved dimension, needed only for irfft 
+function irfft_dim end
+
+mutable struct AdjointPlan{T,P} <: Plan{T}
+    p::P
+    pinv::Plan
+    AdjointPlan{T,P}(p) where {T,P} = new(p)
+    # always have adjoint inside scaled
+    AdjointPlan{T,P}(p::P) where {T,P<:ScaledPlan{T}} = ScaledPlan{T}(AdjointPlan{T}(p.p), p.scale)
```

Comment:
This constructor is misleading - it creates a `ScaledPlan` but not an `AdjointPlan`. In particular inner (but IMO also outer) constructors should create the desired types. I think the API should just be `adjoint(p::Plan)` which by default creates an `AdjointPlan` but can create a different type if needed (such as for `ScaledPlan`s). The constructors for `AdjointPlan` could/should be an internal detail.